### PR TITLE
Add task to create configuration tarballs

### PR DESCRIPTION
**Pull request by andybradshaw**
This change is [Reviewable](https://reviewable.io/reviews/palantir/sls-packaging/263)

### CODE REVIEW

**Comment by iamdanfox** [comment]
@andybradshaw code looks nice and clean but what's the use case for this `configTar` task?  Does it need a README addition too?

**Comment by bmoylan** [comment]
Yeah we should add relevant docs to the readme. @iamdanfox there's an internal spec PR that explains what this is for.

@andybradshaw do we need to wire up any publishing or do consumers do that themselves in their own gradle buildscripts?

**Comment by andybradshaw** [comment]
I thought leaving the publishing up to consumers was fine. Will update the README.

**Comment by andybradshaw** [comment]
I added the task and a little blurb about it to the readme. I'm not sure if it's clear what I mean by `deployment configuration`. Could probably elaborate a little bit more on that.

File: gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/ConfigTarTask.groovy [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+class ConfigTarTask {
+    static Tar createConfigTarTask(Project project, String taskName, String productType) {
+        project.tasks.create(taskName, Tar) { p ->
+            p.group = JavaServiceDistributionPlugin.GROUP_NAME
+            p.description = "Creates a compressed, gzipped tar file that contains the sls configuration files for the product"
+            // Set compression in constructor so that task output has the right name from the start.
+            p.compression = Compression.GZIP
+            // The extension is the product type without the version
+            // service.v1 -> .service.config.tgz
```
Comment (uschi2000) [inline gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/ConfigTarTask.groovy:34]: should assert that the type has this format and fail if it doesn't

Comment (andybradshaw) [reply gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/ConfigTarTask.groovy:34]: I'm fine adding a check for this, but it looks like the distribution extension already enforces that it comes from a predefined list of types: 
```java
void productType(String type) {
    if (!VALID_PRODUCT_TYPES.contains(type)) {
        throw new IllegalArgumentException("Invalid product type '${type}' specified; supported types: ${VALID_PRODUCT_TYPES}.")
    }
    this.productType = type
}
```

Comment (uschi2000) [reply gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/ConfigTarTask.groovy:34]: kind of brittle, but ok

**Comment by andybradshaw** [comment]
Spent a little bit of time looking into why this is flaking so much. It flakes on my machine with some IO errors for `:testCreateManifest`, but with lower frequency.

**Review by iamdanfox — APPROVED** [review]

**Pull request merged**